paasta_tools/vitesscluster_tools.py

```diff
@@ -48,8 +597,153 @@ def __init__(
             branch_dict=branch_dict,
         )
 
-    def get_instances(self, with_limit: bool = True) -> int:
-        return self.config_dict.get("replicas", 1)
+    def get_namespace(self) -> str:
+        return KUBERNETES_NAMESPACE
+
+    def get_images(self) -> Dict[str, str]:
+        vitess_images = self.config_dict.get(
+            "images", load_system_paasta_config().get_vitess_images()
+        )
+        return {
+            "vtctld": vitess_images["vtctld_image"],
+            "vtadmin": vitess_images["vtadmin_image"],
+            "vtgate": vitess_images["vtgate_image"],
+            "vttablet": vitess_images["vttablet_image"],
+        }
+
+    def get_env_variables(self) -> List[Dict[str, Any]]:
+        # get all K8s container env vars and format their keys to camel case
+
+        # Workaround from https://github.com/kubernetes-client/python/issues/390
+        api_client = ApiClient()
+        env = [
+            api_client.sanitize_for_serialization(env)
+            for env in self.get_container_env()
+        ]
+        return env
+
+    def get_labels(self) -> Dict[str, str]:
+        # get default labels from parent class to adhere to paasta contract
+        docker_url = self.get_docker_url(
+            system_paasta_config=load_system_paasta_config()
+        )
+        git_sha = get_git_sha_from_dockerurl(docker_url)
+        labels = self.get_kubernetes_metadata(git_sha=git_sha).labels
+        if "yelp.com/owner" in labels.keys():
+            labels["yelp.com/owner"] = "dre_mysql"
+        return labels
+
+    def get_vitess_node_affinity(self) -> dict:
+        paasta_pool = self.get_pool()
+        self.config_dict.update(
+            {
+                "node_selectors": {
+                    "yelp.com/pool": [paasta_pool],
+                }
+            }
+        )
+
+        # Workaround from https://github.com/kubernetes-client/python/issues/390
+        api_client = ApiClient()
+        node_affinity = api_client.sanitize_for_serialization(self.get_node_affinity())
+        return node_affinity
+
+    def get_region(self) -> str:
+        superregion = self.get_cluster()
+        superregion_to_region_map = (
+            load_system_paasta_config().get_superregion_to_region_mapping()
+        )
+        region = None
+        for superregion_prefix in superregion_to_region_map:
+            if superregion.startswith(superregion_prefix):
+                region = superregion.replace(
+                    superregion_prefix, superregion_to_region_map[superregion_prefix]
+                )
+        if region is None:
+            log.error(
+                f"Region not found for superregion {superregion}. Check superregion_to_region_mapping in system paasta config"
+            )
+        return region
```

Comment:
is it safe to continue if this happens? we're saying get_region will return a str, but it's really returning `Optional[str]`

Reply:
Fixed in https://github.com/Yelp/paasta/pull/3889/commits/2f6ecc76417760dbd1cb424c3b61dabd5c83ea09